### PR TITLE
CC-1313

### PR DESCRIPTION
**Pull request by wwwtyro**

### CODE REVIEW

File: coordinator/client/connection.go [hunk 1]
```diff
@@ -39,12 +39,20 @@ type Node interface {
 	SetVersion(interface{})
 }
 
+type Transaction interface {
+	Create(path string, node Node)
```
Comment (smousa) [inline coordinator/client/connection.go]: Should these calls return error?

Comment (wwwtyro) [inline coordinator/client/connection.go]: If they ever do anything more complicated than they are, yes maybe, but they can't if we're going to try to implement chaining.

**Comment by wwwtyro** [comment]
retest this

File: coordinator/client/zookeeper/connection.go [hunk 2]
```diff
@@ -154,6 +162,31 @@ func (c *Connection) Create(path string, node client.Node) error {
...
+	path = join(c.basePath, path)
+	split := strings.Split(path, "/")
+	final := strings.Join(split[:len(split)-1], "/")
+	exists, err := c.Exists(final)
+	if err != nil {
+		glog.Errorf("Error testing existence of node %s: %s", final, err)
+		return xlateError(err)
+	}
+	if exists {
+		return xlateError(nil)
```
Comment (jplouis) [inline coordinator/client/zookeeper/connection.go]: Can we get better error messages?

File: coordinator/client/zookeeper/connection.go [hunk 3]
```diff
@@ -154,6 +162,31 @@ func (c *Connection) Create(path string, node client.Node) error {
...
+	}
+	_path := ""
+	for _, n := range split[1 : len(split)-1] {
+		_path += "/" + n
+		_, err = c.conn.Create(_path, []byte{}, 0, zklib.WorldACL(zklib.PermAll))
+		if err != nil && err != zklib.ErrNodeExists {
+			return xlateError(err)
+		}
+	}
+	return xlateError(nil)
```
Comment (jplouis) [inline coordinator/client/zookeeper/connection.go]: Can we get better error messages?

File: coordinator/client/zookeeper/connection_test.go [hunk 4]
```diff
@@ -133,6 +133,182 @@ func TestZkDriver(t *testing.T) {
...
+		Name: "test0",
+	}
+	testNode1 := &testNodeT{
+		Name: "test1",
+	}
+
+	multi := conn.NewTransaction()
+	multi.Create("/test0", testNode0)
+	multi.Set("/test1", testNode1)
+	if err = multi.Commit(); err == nil {
```
Comment (jplouis) [inline coordinator/client/zookeeper/connection_test.go]: Do you want to verify the error type/message?

Comment (wwwtyro) [inline coordinator/client/zookeeper/connection_test.go]: I do, but sadly we get a generic error for all transaction failures with this version of go-zookeeper. The up-to-date version breaks it down by transaction item.

**Pull request merged**